File: c/gtk4/label.c
#include <gtk/gtk.h>

void onAppActivate(GApplication *self, gpointer data);
void onAppStartup(GApplication *self, gpointer data);

const gchar *APP_ID = "io.github.Miqueas.GTK-Examples.C.Gtk4.Label";
const gchar *APP_TITLE = "GtkLabel";

int main(int argc, char **argv) {
  const gchar *APP_ID = "io.github.Miqueas.GTK-Examples.C.Gtk4.Label";
  GtkApplication *app = gtk_application_new(APP_ID, 0);

  g_signal_connect(app, "startup",  G_CALLBACK(onAppStartup), NULL);
  g_signal_connect(app, "activate", G_CALLBACK(onAppActivate), NULL);

  int result = g_application_run(G_APPLICATION(app), argc, argv);
  g_object_unref(app);

  return result;
}

void onAppActivate(GApplication *self, gpointer data) {
  GtkWindow *window = gtk_application_get_active_window(GTK_APPLICATION(self));
  gtk_window_present(window);
}

void onAppStartup(GApplication *self, gpointer data) {
  GtkWidget *window, *label;

  window = gtk_application_window_new(GTK_APPLICATION(self));
  label = gtk_label_new("Hi there!");

  gtk_window_set_child(GTK_WINDOW(window), label);
  gtk_window_set_title(GTK_WINDOW(window), APP_TITLE);
  gtk_window_set_default_size(GTK_WINDOW(window), 400, 400);
}

File: lua/gtk4/label.lua
local lgi = require("lgi")
local Gtk = lgi.require("Gtk", "4.0")
local Gio = lgi.Gio

-- GtkLabel: A text widget

local appID = "io.github.Miqueas.GTK-Examples.Lua.Gtk4.Label"
local appTitle = "GtkLabel"
local app = Gtk.Application.new(appID, Gio.ApplicationFlags.FLAGS_NONE)

function app:on_startup()
  local window = Gtk.ApplicationWindow.new(self)
  local label = Gtk.Label.new("Hi there!")

  window.child = label
  window.title = appTitle
  window:set_default_size(400, 400)

  label.valign = Gtk.Align.CENTER
  label.halign = Gtk.Align.CENTER
end

function app:on_activate()
  self.active_window:present()
end

return app:run(arg)